### PR TITLE
Add ORC file input support

### PR DESCRIPTION
**Pull request by bradyoo**

### CODE REVIEW

**Comment by bradyoo** [comment]
alluxio-bot, test this please.

**Comment by bradyoo** [comment]
ping? @gpang

**Comment by bradyoo** [comment]
ping? @gpang

File: job/server/src/main/java/alluxio/job/plan/transform/format/SchemaConversionUtils.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import org.apache.avro.SchemaBuilder;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Utility Class for converting schema to Parquet.
+ */
+public class SchemaConversionUtils {
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/transform/format/SchemaConversionUtils.java:30]: were these methods mainly copied from the previous location?

File: job/server/src/main/java/alluxio/job/plan/transform/format/orc/OrcReader.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ */
+public final class OrcReader implements TableReader {
+
+  private final Closer mCloser;
+  private final OrcSchema mSchema;
+  private final Reader mReader;
+  private final RecordReader mRows;
+  private final List<String> mFieldNames;
+
+  private VectorizedRowBatch mCurrentBatch;
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/transform/format/orc/OrcReader.java:42]: can you add javadoc comments for these 2 members?

File: job/server/src/main/java/alluxio/job/plan/transform/format/orc/OrcSchema.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  private final ArrayList<FieldSchema> mAlluxioSchema;
+
+  private final Schema mWriteSchema;
+
+  /**
+   * Default constructor for OrcSchema.
+   * @param reader the orc reader
+   */
+  public OrcSchema(Reader reader) throws IOException {
+    mAlluxioSchema = new ArrayList<>();
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/transform/format/orc/OrcSchema.java:40]: can this be initialized with the size?

File: job/server/src/main/java/alluxio/job/plan/transform/format/SchemaConversionUtils.java (hunk 1 above)
Comment (bradyoo) [reply job/server/src/main/java/alluxio/job/plan/transform/format/SchemaConversionUtils.java:30]: Yep. These were mainly copied

**Comment by bradyoo** [comment]
reflected feedback

**Comment by bradyoo** [comment]
Ping? @gpang

**Comment by bradyoo** [comment]
I added a single OrcReader test that relies on reading a new resource file job/server/src/test/resources/TestOrcFile.columnProjection.orc which is from the apache's Orc repo. Let me know if that's fine.

**Comment by bradyoo** [comment]
Ping? @gpang

**Comment by bradyoo** [comment]
Ping? @gpang

**Comment by bradyoo** [comment]
Can you merge this for me? @gpang

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**